modules/dists/CyclicDist.chpl

```diff
@@ -373,8 +353,19 @@ override proc Cyclic.dsiNewRectangularDom(param rank: int, type idxType, param s
     compilerError("Cyclic domain index type does not match distribution's");
   if rank != this.rank then
     compilerError("Cyclic domain rank does not match distribution's");
-  var dom = new unmanaged CyclicDom(rank=rank, idxType=idxType, dist = _to_unmanaged(this), stridable=stridable);
-  dom.dsiSetIndices(inds);
+  const whole = createWholeDomainForInds(rank, idxType, stridable, inds);
+
+  const dummyLCD = new unmanaged LocCyclicDom(rank, idxType);
```

Comment:
Also seems like it could be part of an initializer.

Are the ``dsiSetIndices`` calls being removed because we're now passing 'whole' in?

Reply:
The same discussion as for `BlockDom` above.

> Are the `dsiSetIndices` calls being removed because we're now passing 'whole' in?

Correct. The `dsiSetIndices` call is now replaced by calling `createWholeDomainForInds` then setting up `locDoms` array explicitly. The latter used to be done in `setup()` that was, in turn, invoked from `dsiSetIndices`.

All these changes are due to a switch to make `locDoms` non-nilable -- correctly -- when it is initializes as part of its enclosing `BlockDom` object.

Alternatively, we could change `locDoms` to contain nilable elements and add postfix-`!` everywhere that its elements are referenced. This would allow us to revert most of the changes that this PR makes to domain maps.